src/java.base/share/classes/jdk/internal/vm/VMSupport.java

```diff
@@ -51,32 +56,56 @@ public static synchronized Properties getAgentProperties() {
     private static native Properties initAgentProperties(Properties props);
 
     /**
-     * Write the given properties list to a byte array and return it. Properties with
-     * a key or value that is not a String is filtered out. The stream written to the byte
-     * array is ISO 8859-1 encoded.
+     * Writes the given properties list to a byte array and return it. The stream written
+     * to the byte array is ISO 8859-1 encoded.
+     *
+     * @param filter if true, then entries in {@code p} with a key or value that is not a
+     *               String are filtered out. Otherwise, the caller guarantees {@code p}
+     *               only contains String keys and values.
      */
-    private static byte[] serializePropertiesToByteArray(Properties p) throws IOException {
+    private static byte[] serializePropertiesToByteArray(Properties p, boolean filter) throws IOException {
         ByteArrayOutputStream out = new ByteArrayOutputStream(4096);
 
-        Properties props = new Properties();
+        Properties props;
+        if (filter) {
+            props = new Properties();
 
-        // stringPropertyNames() returns a snapshot of the property keys
-        Set<String> keyset = p.stringPropertyNames();
-        for (String key : keyset) {
-            String value = p.getProperty(key);
-            props.put(key, value);
+            // stringPropertyNames() returns a snapshot of the property keys
+            Set<String> keyset = p.stringPropertyNames();
+            for (String key : keyset) {
+                String value = p.getProperty(key);
+                props.put(key, value);
+            }
+        } else {
+            props = p;
         }
 
         props.store(out, null);
         return out.toByteArray();
     }
 
     public static byte[] serializePropertiesToByteArray() throws IOException {
-        return serializePropertiesToByteArray(System.getProperties());
+        return serializePropertiesToByteArray(System.getProperties(), true);
     }
 
     public static byte[] serializeAgentPropertiesToByteArray() throws IOException {
-        return serializePropertiesToByteArray(getAgentProperties());
+        return serializePropertiesToByteArray(getAgentProperties(), true);
+    }
+
+    /**
+     * Serializes {@link VM#getSavedProperties()} to a byte array.
+     *
+     * Used by JVMCI to copy properties into libjvmci.
+     */
+    public static byte[] serializeSavedPropertiesToByteArray() throws IOException {
+        Properties props = new Properties();
+        for (var e : VM.getSavedProperties().entrySet()) {
+            props.put(e.getKey(), e.getValue());
+        }
+        if (props.get("oome") != null) {
+            throw new OutOfMemoryError("forced OOME");
```

Comment:
I don't think code in java.base should be checking for a property named "oome".  Is this for a test that sets this system property on the command line?

Reply:
Sorry, that's debug code that I will remove.

Reply:
Thanks, I couldn't see it set as a property anywhere so didn't know why it was there. I don't have any comments on this area.